contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -393,23 +390,35 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
      */
     function sendSpecifiedReceipts(
         bytes32 originBlockchainID,
-        uint256[] calldata messageIDs,
+        bytes32[] calldata messageIDs,
         TeleporterFeeInfo calldata feeInfo,
         address[] calldata allowedRelayerAddresses
-    ) external senderNonReentrant returns (uint256) {
+    ) external senderNonReentrant returns (bytes32) {
+        bytes32 blockchainID_ = _initializeBlockchainID();
```

Comment:
same here, think `blockchainID` must be initialized to pass the receipt found check, since we need to receive before